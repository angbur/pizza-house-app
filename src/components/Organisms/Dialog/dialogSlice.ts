import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { RootState } from 'store/store';

export enum FormType {
  register = 'register',
  login = 'login',
  orderDetail = 'orderDetail',
}

type DialogState = {
  isOpen: boolean;
  formType: FormType | undefined;
};

const initialState: DialogState = {
  isOpen: false,
  formType: undefined,
};

const dialogSlice = createSlice({
  name: 'dialog',
  initialState,
  reducers: {
    openDialog: (state, action: PayloadAction<FormType>) => {
      state.isOpen = true;
      state.formType = action.payload;
    },
    closeDialog: (state) => {
      state.isOpen = false;
      state.formType = undefined;
    },
  },
});

export const { openDialog, closeDialog } = dialogSlice.actions;

export default dialogSlice.reducer;

export const selectDialogState = (state: RootState) => state.dialog;
