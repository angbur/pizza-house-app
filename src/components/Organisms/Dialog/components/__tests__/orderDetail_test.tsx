import { renderWithProviders } from 'test/test-utils';
import { getDialogElement } from '../../dialog.utils';
import { FormType } from '../../dialogSlice';

describe('Register content of dialog', () => {
  it('should renders register component correctly', () => {
    const initialState = {
      loggedIn: true,
      status: 200,
      error: undefined,
      userId: 'dummy_userId',
      login: 'dummy_login',
      token: 'dummy_token',
    };
    const handleChange = jest.fn();
    const dialog = () => (
      <>{getDialogElement(FormType.orderDetail)?.component({ handleChange: handleChange })}</>
    );

    const orderDetail = renderWithProviders(dialog(), {
      preloadedState: {
        user: initialState,
      },
    }).container;

    expect(orderDetail).toMatchSnapshot();
  });
});
