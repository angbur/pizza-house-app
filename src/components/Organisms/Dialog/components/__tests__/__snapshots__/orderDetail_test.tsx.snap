// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register content of dialog should renders register component correctly 1`] = `
<div>
  <div
    class="sc-dkrFOg keXQyv"
  >
    <h5
      class="item-title"
      style="font-family: Source Sans Pro; font-weight: 600; font-size: 1rem; line-height: 1.4; letter-spacing: 1px; color: rgb(21, 21, 21);"
    >
      In progress...
       
    </h5>
  </div>
</div>
`;
